src/migration/MigrationExecutor.ts

```diff
@@ -72,9 +85,14 @@ export class MigrationExecutor {
         // find all migrations that needs to be executed
         const pendingMigrations = allMigrations.filter(migration => {
             // check if we already have executed migration
-            const executedMigration = executedMigrations.find(executedMigration => executedMigration.name === migration.name);
-            if (executedMigration)
+            const executedMigration = executedMigrationsByName[migration.name];
+            if (executedMigration) {
+                delete executedMigrationsByName[migration.name];
+                if (!this.migrationsForce && executedMigration.hash !== migration.hash) {
+                    throw new Error(`Migration hash for ${executedMigration.name} does not match: ${executedMigration.hash} !== ${migration.hash}`);
```

Comment:
Add some info about force option? Something like "Use force if you want to skip this check". Maybe better message

Reply:
Thanks, I just pushed a change.